pkg/clusterversion/clusterversion.go

```diff
@@ -90,3 +98,248 @@ func (builder *Builder) validate() (bool, error) {
 
 	return true, nil
 }
+
+// WithDesiredUpdateImage adds the desired image to the clusterversion struct.
+func (builder *Builder) WithDesiredUpdateImage(desiredUpdateImage string, force bool) (*Builder, error) {
+	var err error
```

Comment:
I think you can safely remove this line.

Reply:
Yes because with function can return builder only. Please remove error from return and use builder.errorMsg instead. Please refer to: https://github.com/openshift-kni/eco-goinfra/blob/4b536d23d57aedc397057a32d67fd2823296326e/pkg/sriov/network.go#L105